packages/communication-react/review/communication-react.api.md

```diff
@@ -253,8 +262,30 @@ export interface CallAdapter {
     on(event: 'callEnded', listener: CallEndedListener): void;
     // (undocumented)
     on(event: 'error', listener: (e: Error) => void): void;
+}
+
+// @public
+export type CallAdapterClientState = {
+    userId: CommunicationUserKind;
+    displayName?: string;
+    call?: CallState;
+    devices: DeviceManagerState;
+};
+
+// @public (undocumented)
+export interface CallAdapterHandlers {
```

Comment:
Recommend against the `Handler` and `Selector` terminology in this API. The adapter API is disassociated from the stateful API and those words don't mean anything in this context. Using those words here will make it harder to keep these API layers conceptually separate.

Reply:
How about `CallAdapterCallManagement` and `CallAdapterDeviceManagement`. The interfaces are then more aligned with `@azure/communication-calling`, but Contoso gets to use all the interfaces from a single exported object. So we get the best of both worlds.

Reply:
Yeah I agree handlers isn't the best term, I'll update to Call/Device/ChatManagement

Reply:
Do you think the likes of startCamera and setCamera would fall under DeviceManagement or CallManagement?

Reply:
Thinking: 
```
export interface CallAdapterCallManagement {
  joinCall(microphoneOn?: boolean): Call | undefined;
  leaveCall(forEveryone?: boolean): Promise<void>;
  startCamera(): Promise<void>;
  stopCamera(): Promise<void>;
  onToggleCamera(options?: VideoStreamOptions): Promise<void>;
  mute(): Promise<void>;
  unmute(): Promise<void>;
  startCall(participants: string[]): Call | undefined;
  startScreenShare(): Promise<void>;
  stopScreenShare(): Promise<void>;
  removeParticipant(userId: string): Promise<void>;
  createStreamView(remoteUserId?: string, options?: VideoStreamOptions): Promise<void>;
  disposeStreamView(remoteUserId?: string, options?: VideoStreamOptions): Promise<void>;
}

export interface CallAdapterDeviceManagement {
  askDevicePermission(constrain: PermissionConstraints): Promise<void>;
  queryCameras(): Promise<VideoDeviceInfo[]>;
  queryMicrophones(): Promise<AudioDeviceInfo[]>;
  querySpeakers(): Promise<AudioDeviceInfo[]>;
  setCamera(sourceId: VideoDeviceInfo, options?: VideoStreamOptions): Promise<void>;
  setMicrophone(sourceId: AudioDeviceInfo): Promise<void>;
  setSpeaker(sourceId: AudioDeviceInfo): Promise<void>;
}
```

Reply:
Updated to reflect this